department-of-veterans-affairs/vets-website
# Update fields

## Summary
Update dependent child information pages to use web component patterns where applicable on the Pension Benefits form (21P-527EZ)

### Note
The additional information component was moved below the 'Is your child seriously disabled?' question until the 'yesNo' web component pattern has description support.

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Use the 'SIP' menu to load the form with maximal-data.json
3. Go to the 'Dependent information' pages in the 'Household information' chapter
4. Verify that the 'in school' and 'is disabled' questions are visible/hidden based on dependent's age
5. Verify that the fields are validated when required
6. Verify that the data is saved to the api and redux
7. Go to the 'Review application' page
8. Verify that the 'Dependent information' data is viewable and editable
9. Verify that the application is able to be submitted

## Acceptance criteria
[Issue in Github - 78678](https://github.com/department-of-veterans-affairs/va.gov-team/issues/78678)

## Related issue(s)
n/a

## Screenshots
| Before | After |
| ------ | ------ |
|![localhost_3001_pension_application_527EZ_household_dependents_children_information_2](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/fc62e897-f91f-452c-877d-15a7a4e05a98)|![localhost_3001_pension_application_527EZ_form-saved (21)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/264d0f46-af03-48b3-800a-103a648b4ef0)


## What areas of the site does it impact?
Pension Benefits Application

### Quality Assurance & Testing

### Error Handling

### Authentication
